### PR TITLE
BD-52 feat: add sidebar

### PR DESCRIPTION
**Pull request by pRomKol**

### CODE REVIEW

File: components/navbar/NavBar.tsx
Comment (sfgardev) [inline components/navbar/NavBar.tsx:1]: rename file to Navbar

File: components/navbar/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import { useAppDispatch } from '@/store'
+import Link from 'next/link'
+import { Button, Icon, SideBar, Typography } from 'uikit-inctagram'
+
+type Menu = {
+  href: string
+  icon: ReactNode
+  label: string
+}
+type Props = ComponentProps<'aside'>
```
Comment (sfgardev) [inline components/navbar/NavBar.tsx:16]: why do we need props here?

File: components/navbar/NavBar.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  const { data } = authApi.useAuthMeQuery()
+  const [isModalOpen, setIsModalOpen] = useState(false)
+  const dispatch = useAppDispatch()
+  const [logout, { isLoading: isLogoutLoading }] = authApi.useLogoutMutation()
+
+  const handleOpenModal = () => setIsModalOpen(true)
+  const handleCloseModal = () => setIsModalOpen(false)
+
+  const handleLogout = () => {
+    startTransition(() => {
```
Comment (sfgardev) [inline components/navbar/NavBar.tsx:29]: for what purpose you use startTransition?

File: components/navbar/NavBar.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      href: '#',
+      icon: <Icon icon='trending-up-outline' />,
+      label: t.navbar.statistic,
+    },
+    {
+      href: '#',
+      icon: <Icon icon='bookmark-outline' />,
+      label: t.navbar.favorites,
+    },
+  ]
```
Comment (sfgardev) [inline components/navbar/NavBar.tsx:80]: it would be better to move these menus to separate file

File: components/navbar/NavBar.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+  return (
+    <SideBar className={className}>
```
Comment (sfgardev) [inline components/navbar/NavBar.tsx:83]: add styles here, instead of passing className

File: components/navbar/NavBar.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      <ConfirmationModal
```
Comment (sfgardev) [inline components/navbar/NavBar.tsx:84]: you can move ConfirmationModal out of Sidebar by wrapping all by fragment

File: public/locales/en.ts [hunk 6]
```diff
@@ -78,5 +79,17 @@ export const en = {
       return `Are you really want to log out of your account ${email}?`;
     },
     ru: "Russian"
+  },
+  navbar: {
+    button: {
```
Comment (sfgardev) [inline public/locales/en.ts:84]: if you add logout to button, remove it from here

**Review by sfgardev — CHANGES_REQUESTED** [review]

**Comment by sfgardev** [comment]
image

File: components/navbar/Navbar.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import { logoutIcon, menu1, menu2 } from '@/components/navbar/constans'
+import { authActions } from '@/features'
+import { useTranslation } from '@/hooks/useTranslation'
+import { LocaleType } from '@/public'
+import { authApi } from '@/services'
+import { useAppDispatch } from '@/store'
+import Link from 'next/link'
+import { Button, SideBar, Typography } from 'uikit-inctagram'
+
+export type MenuItem = {
```
Comment (sfgardev) [inline components/navbar/Navbar.tsx:13]: move to constants

File: components/navbar/Navbar.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        onClose={handleCloseModal}
+        onConfirm={handleLogout}
+        title='Confirm logout'
+      />
+      <SideBar className='h-full flex-col items-center justify-start pb-9 pt-[72px]'>
+        <nav className='flex h-full flex-col'>
+          <ul className='flex flex-col gap-6'>{renderMenuItems(menu1)}</ul>
+          <ul className='mt-[60px] flex flex-col gap-6'>{renderMenuItems(menu2)}</ul>
+          <Button
+            className='mt-auto flex items-center gap-3 p-0 font-medium text-light-100' // Добавьте временный фон для отладки
```
Comment (sfgardev) [inline components/navbar/Navbar.tsx:82]: remove comment

File: components/navbar/Navbar.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+      .catch(err => {
+        console.error(err)
+      })
+  }
+
+  const renderMenuItems = (menuItems: MenuItem[]) => {
+    const translatedMenuItems = getMenuItemsWithTranslation(menuItems, t)
+
+    return translatedMenuItems.map(menuItem => (
+      <li className='text-light-100' key={menuItem.label}>
```
Comment (sfgardev) [inline components/navbar/Navbar.tsx:51]: fix ts errors

**Review by sfgardev — CHANGES_REQUESTED** [review]

File: components/navbar/constans.tsx
Comment (sfgardev) [inline components/navbar/constans.tsx:1]: change name to constants

File: components/navbar/Navbar.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  return menuItems.map(menuItem => ({
+    ...menuItem,
+    label: locale.navbar[menuItem.translationId],
+  }))
+}
+
+export const Navbar = () => {
+  const router = useRouter()
+  const [isModalOpen, setIsModalOpen] = useState(false)
+  const [activeLink, setActiveLink] = useState(router.pathname)
```
Comment (sfgardev) [inline components/navbar/Navbar.tsx:24]: useState is redundant here, instead use usePathname hook
https://medium.com/@theambi/how-to-easily-style-active-link-in-nextjs-ebd01a484f23 see example here

File: components/navbar/Navbar.tsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+      })
+  }
+
+  const renderMenuItems = (menuItems: MenuItem[]) => {
+    const translatedMenuItems = getMenuItemsWithTranslation(menuItems, t)
+
+    return translatedMenuItems.map(menuItem => (
+      <li className='text-light-100' key={menuItem.label}>
+        <Link
+          className={`flex items-center gap-3 transition ${activeLink === menuItem.href ? 'text-accent-100' : 'hover:text-accent-100 active:text-accent-700'}`}
```
Comment (sfgardev) [inline components/navbar/Navbar.tsx:49]: use cn for dynamic styles

**Review by sfgardev — CHANGES_REQUESTED** [review]

File: components/navbar/Navbar.tsx (hunk 10 above)
Comment (pRomKol) [reply components/navbar/Navbar.tsx:24]: done

File: components/navbar/Navbar.tsx (hunk 11 above)
Comment (pRomKol) [reply components/navbar/Navbar.tsx:49]: done

File: components/navbar/constans.tsx
Comment (pRomKol) [reply components/navbar/constans.tsx:1]: done

**Review by sfgardev — APPROVED** [review]

**Pull request merged**